### PR TITLE
Add parameters

### LINKED ISSUE #150: Extract model calculations from app
Refactor to display_sidebar and push calculations into Parameters class in models.py

### PR DESCRIPTION
**Pull request by jlubken**
Pulls out most all of the globals.

Extract sidebar into function to be moved elsewhere.
Extract calculations into models:Parameters class.

Closed #150 and #111

### CODE REVIEW

File: penn_chime/models.py [hunk 1]
```diff
@@ -1,9 +1,118 @@
...
+        self.rates = tuple(
+            each.rate
+            for each in (hospitalized, icu, ventilated)
+        )
+        self.lengths_of_stay = tuple(
+            each.length_of_stay
+            for each in (hospitalized, icu, ventilated)
+        )
+
+        # TODO: I tried casting this to an int but this breaks unittest comparisons (third digit)
```
Comment (sam-writer) [inline penn_chime/models.py:47]: IIUC sounds like you could fix the test? those were added with the thinking "something is better than nothing" - not meant to be The Source Of Truth

File: penn_chime/models.py [hunk 2]
```diff
@@ -1,9 +1,118 @@
+from collections import namedtuple
 from typing import Generator, Tuple
 
 import numpy as np
 import pandas as pd
 import streamlit as st
 
+RateLos = namedtuple('RateLos', ('rate', 'length_of_stay'))
```
Comment (quinn-dougherty) [inline penn_chime/models.py]: can this be `from defaults import RateLos` instead?

File: penn_chime/models.py [hunk 3]
```diff
@@ -1,9 +1,118 @@
...
+
+        # Simplify equation to avoid division by zero:
+        # self.r_naught = r_t / (1.0 - relative_contact_rate)
+        self.r_naught = (intrinsic_growth_rate + gamma)  / gamma
+
+        # doubling time after distancing
+        # TODO constrain values np.log2(...) > 0.0
+        self.doubling_time_t = 1.0 / np.log2(beta * susceptible - gamma + 1)
+
+        self.beta_decay = 0.0
```
Comment (sam-writer) [inline penn_chime/models.py:85]: This should be a parameter to `__init__` IMO, defaulting to this value

File: penn_chime/models.py [hunk 4]
```diff
@@ -1,9 +1,118 @@
...
+            self.detection_probability = None
+
+        # TODO missing initial recovered value
+        self.recovered = 0.0
+
+        self.intrinsic_growth_rate = intrinsic_growth_rate = \
+            2.0 ** (1.0 / doubling_time) - 1.0 if doubling_time > 0.0 else 0.0
+
+        # TODO make this configurable, or more nuanced
+        self.recovery_days = recovery_days = 14.0
```
Comment (sam-writer) [inline penn_chime/models.py:64]: This should be a parameter to `__init__` IMO, defaulting to this value

File: penn_chime/models.py [hunk 5]
```diff
@@ -1,9 +1,118 @@
...
+        self.infected = infected = \
+            current_hospitalized / market_share / hospitalized.rate
+
+        if infected > 1.e-7:
+            self.detection_probability = known_infected / infected
+        else:
+            self.detection_probability = None
+
+        # TODO missing initial recovered value
+        self.recovered = 0.0
```
Comment (sam-writer) [inline penn_chime/models.py:58]: This should be a parameter to `__init__` IMO, defaulting to this value

File: penn_chime/models.py (hunk 1 above)
Comment (quinn-dougherty) [reply penn_chime/models.py:47]: i second sam's comment

File: penn_chime/models.py (hunk 2 above)
Comment (jlubken) [reply penn_chime/models.py]: Of course, I just noticed that.

File: penn_chime/models.py (hunk 1 above)
Comment (jlubken) [reply penn_chime/models.py:47]: That comment was from PJ's merge.

**Review by quinn-dougherty — APPROVED** [review]
very excellent. well done.

File: penn_chime/models.py (hunk 1 above)
Comment (jlubken) [reply penn_chime/models.py:47]: Actually, infected shouldn't be cast to an int here, it is a computed guess based off of the other numbers we do have.

**Review by sam-writer — COMMENTED** [review]
Well I am not fast enough

**Pull request merged**